demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -82,60 +84,75 @@
     #define NETWORK_BUFFER_SIZE    ( 1024U )
 #endif
 
+/**
+ * @brief The largest random number to use in client identifier.
+ *
+ * @note Random number is added to MQTT client identifier to avoid client
+ * identifier collisions while connecting to MQTT broker.
+ */
+#define MAX_RAND_NUMBER_FOR_CLIENT_ID           ( 999u )
```

Comment:
Seeding considerations aside, 1/1000 isn't particularly good. If a customer were to take this demo and run with it as an application, at about 37-38 demos, there is a 50% chance of a collision.

Not being deterministic across starts of the same binary also seems problematic to a lot of customer use cases.

Reply:
Yes, this is called the "brithday attack", well spotted!